lib/core/include/irods/replica.hpp

```diff
@@ -24,6 +24,7 @@
 #endif
 
 #include "irods/filesystem.hpp"
+#include "irods/filesystem/detail.hpp" // FIXME This shouldn't be used outside of the fsys lib.
```

Comment:
See FIXME comment

Reply:
The question now is how do I address this? Hmm.

Seems my only option is to introduce another file and move the functions into it. Maybe `irods/filesystem/path_utilities.hpp`? Or maybe the detail header becomes `path_utilities.hpp`.

Reply:
The only problem with that is if someone modifies the behavior in the future and it results in a change in behavior for the fsys lib.

Now, if the replica lib is an extension of the fsys lib, then maybe we don't have to change anything.

Thoughts?

Reply:
The tests for these libraries aren't exhaustive as far as exceptions are concerned. How would we detect a behavior change before release?

The other thing is that having a detail namespace or header for a lib is signal that those types/functions/etc are private to the implementation and shouldn't be used. I feel we have to separate them to future proof ourselves.

I really don't want to duplicate work, but maybe that is okay here because the functions we're talking about aren't complex.

Reply:
I feel that a shared space for these types of functions would still be good. Why are these functions in the detail namespace? I don't think I understand the concern with exposing them in something like `path_utilities.hpp`.

Reply:
After some discussion, the plan is to move the functions into a new header file, `filesystem/path_utilities.hpp`.

Reply:
Done.